packages/manager/apps/web/client/app/domain/webhosting/order/steps/domain-webhosting-order-steps.controller.js

```diff
@@ -10,15 +17,57 @@ export default class {
   }
 
   orderWebhosting() {
-    this.validatingCheckout = true;
-    return this.validateCheckout(this.cartId, {
-      autoPayWithPreferredPaymentMethod: this.autoPayWithPreferredPaymentMethod,
-    }).catch(() =>
-      this.displayErrorMessage(
-        this.$translate.instant(
-          'domain_webhosting_order_payment_checkout_error',
-        ),
-      ),
+    const getDnsZoneLabel = () => {
+      const { enableHosting, enableEmails } = this.cartOption.dnsConfiguration;
+      if (enableHosting && enableEmails) {
+        return CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.RESET_ALL;
+      }
+      if (enableHosting) {
+        return CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.RESET_ONLY_A;
+      }
+      if (enableEmails) {
+        return CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.RESET_ONLY_MX;
+      }
+      return CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.NO_CHANGE;
+    };
+    const dnsZoneLabel = getDnsZoneLabel();
```

Comment:
Maybe you could drop this function and directly set the value:
```suggestion
    const { enableHosting, enableEmails } = this.cartOption.dnsConfiguration;
    let dnsZoneLabel = CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.NO_CHANGE;
      if (enableHosting && enableEmails) {
        dnsZoneLabel = CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.RESET_ALL;
      }
      if (enableHosting) {
        dnsZoneLabel = CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.RESET_ONLY_A;
      }
      if (enableEmails) {
        dnsZoneLabel = CONFIGURATION_OPTIONS.DNS_ZONE.VALUES.RESET_ONLY_MX;
      }
```

Reply:
Seems good too :)